test/controllers/integrations/github_controller_test.rb

```diff
@@ -53,6 +53,31 @@
       request.headers['X-Hub-Signature'] = "sha1=#{hmac}"
     end
 
-    test_regular_commit "Github", no_mapping: { ref: 'refs/heads/origin/foobar' }, failed: false
+    test_regular_commit "Github", no_mapping: { ref: 'refs/heads/foobar' }, failed: false
+  end
+
+  describe 'security' do
+    let(:user_name) { 'Github' }
+
+    before do
+      project.webhooks.each(&:soft_delete!)
+      request.headers['X-Github-Event'] = 'push'
+      @payload = payload.deep_merge(token: project.token)
+      @hmac = OpenSSL::HMAC.hexdigest(OpenSSL::Digest.new('sha1'), 'test', payload.to_param)
+    end
+
+    it 'is timing safe' do
+      request.headers['X-Github-Event'] == 'invalid'
+      request.headers['X-Hub-Signature'] = "sha1=#{@hmac}"
+      time_success = Benchmark.measure do
+        1000.times { post :create, @payload }
+      end.total
+      request.headers['X-Hub-Signature'] = "sha1=#{@hmac.sub(/[0-8a-f]/, '9')}"
+      time_fail = Benchmark.measure do
+        1000.times { post :create, @payload }
+      end.total
+      time_difference = (time_success - time_fail).abs
+      pending { assert time_difference < 0.002 }
```

Comment:
ah it's also pending ... maybe kill the test ... and add a comment on why we are doing the secure-compare ... maybe using secure-compare is unnecessary/useless anyway since there is no way to detect a string compare diff in a 100ms response ...